cmd/createtree/main.go

```diff
@@ -144,9 +139,31 @@ func newRequest(opts *createOpts) (*trillian.CreateTreeRequest, error) {
 		SignatureAlgorithm: sigpb.DigitallySigned_SignatureAlgorithm(sa),
 		DisplayName:        opts.displayName,
 		Description:        opts.description,
-		PrivateKey:         pk,
 		MaxRootDuration:    ptypes.DurationProto(opts.maxRootDuration),
 	}}
+
+	if opts.privateKeyType != "" {
+		pk, err := newPK(opts)
+		if err != nil {
+			return nil, err
+		}
+		ctr.Tree.PrivateKey = pk
+	} else {
+		// Cannot continue if options specifying a key were provided but
+		// privateKeyType is not set, as there's no way to know what protobuf
+		// message type was intended.
+		if opts.pemKeyPath != "" || opts.pemKeyPass != "" || opts.pkcs11ConfigPath != "" {
```

Comment:
This is a little confusing. If we want to generate a key, doesn't that take precedence over the other parameters?  Perhaps issue a warning that these options will be ignored?